synfig-studio/plugins/lottie-exporter/common/Gradient.py

```diff
@@ -0,0 +1,211 @@
+    Currently, it is assumed that only 2 colors will be provided in the gradient
+    """
+
+    def __init__(self, gradient):
+        """
+        Args:
+            gradient (lxml.etree._Element) : Gradient as extracted from the xml
+
+        Returns:
+            (None)
+        """
+        self.synfig_gradient = gradient
+        self.colors = []
+        self.extract_colors()
+
+    def extract_colors(self):
+        """
+        Helps in extracting all the colors present in the gradient
+        """
+        initial_list = []
+        for col in self.synfig_gradient:
+            red = float(col[0].text) ** (1/settings.GAMMA[0])
+            green = float(col[1].text) ** (1/settings.GAMMA[1])
+            blue = float(col[2].text) ** (1/settings.GAMMA[2])
+            alpha = float(col[3].text)
+            initial_list.append([float(col.attrib["pos"]), Color(red, green, blue, alpha)]) 
+        self.colors = initial_list
+
+    def reverse_gamma(self, color):
+        """
+        Given a color, it reverses the effect of gamma as done in extract_colors()
+
+        Args:
+            color (common.Color.Color) : color element
+
+        Returns:
+            (common.Color.Color) : color element with gamma effect reversed
+        """
+        ret = copy.deepcopy(color)
+        ret.red = ret.red ** settings.GAMMA[0]
+        ret.green = ret.green ** settings.GAMMA[1]
+        ret.blue = ret.blue ** settings.GAMMA[2]
+        return ret
+
+    def get_lottie_final_list(self):
+        """
+        Appends the avg of colors as required by lottie format. This function is not currently used
+        anywhere. But might be helpful in future
+
+        Args:
+            (None)
+
+        Returns:
+            (list) : list with average of colors appended
+        """
+        initial_list = copy.deepcopy(self.colors)
+        avg_list = []
+        # Appending the avg colors, as required by lottie
+        for i in range(len(initial_list)-1):
+            col1, col2 = initial_list[i], initial_list[i+1]
+            avg_pos = (col1[0] + col2[0]) / 2
+            avg_col = (col1[1] + col2[1]) / 2
+            avg_list.append([avg_pos, avg_col])
+
+        ret = [None]*(len(initial_list) + len(avg_list))
+        ret[::2] = initial_list
+        ret[1::2] = avg_list
+        return ret
+
+    def get_colors(self):
+        """
+        Returns the list of colors in a gradient
+
+        Args:
+            (None)
+
+        Returns:
+            (list) : list of colors of whom gradient is comprised
+        """
+        return self.colors
+
+
+    def get_val(self):
+        """
+        Get the gradient in the format required by lottie
+
+        Args:
+            (None)
+
+        Returns:
+            (list) : Stores the gradient in the list format
+        """
+        ret = []
+        for col in self.colors:
+            pos = col[0]
+            r, g, b = col[1].red, col[1].blue, col[1].green
```

Comment:
I am getting some different results in the colors. It should have been like this:
```
r, g, b = col[1].red, col[1].green, col[1].blue
```

Reply:
Fixed it